### PR TITLE
feat(helm-docs): autogenerate helm docs

### PR DESCRIPTION
**Pull request by tiagolobocastro**
ci: add helm-docs Jenkins test

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore: update values.yaml and update README

Add comments notation and tidy up some values.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

feat(helm-docs): autogenerate helm docs

Add helm-docs to nix-shell.
Add template for generating the README.
Add script that generates the README.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

chore(shell): add commitlinter as git pre-commit hook

Add the commitlinter as a pre-commit hook

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

ci: run on any agent

We don't need hugepages or nvme-tcp driver so any agent will do.

Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

**Comment by tiagolobocastro** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors try

**Comment by niladrih** [comment]
Helm-docs in question is https://github.com/norwoodj/helm-docs v1.11.0

**Comment by tiagolobocastro** [comment]
> Helm-docs in question is https://github.com/norwoodj/helm-docs v1.11.0

Yes it's that one but v1.8.1

File: chart/values.yaml [hunk 1]
```diff
@@ -396,7 +399,7 @@ loki-stack:
           - mountPath: /data
             name: storage
     config:
-      # Compactor is a MayastorDB(loki term) Shipper specific service that reduces the index
+      # Compactor is a BoltDB(loki term) Shipper specific service that reduces the index
```
Comment (pchandra19) [inline chart/values.yaml:402]: Should we refer as Bolt in openebs ?

**Review by pchandra19 — APPROVED** [review]

File: chart/values.yaml (hunk 1 above)
Comment (tiagolobocastro) [reply chart/values.yaml:402]: BoltDB is the actual name of the loki service.

Comment (pchandra19) [reply chart/values.yaml:402]: ok

Comment (avishnu) [reply chart/values.yaml:402]: Can we not change to MayastorDB here and wherever applicable?

File: chart/values.yaml [hunk 2]
```diff
@@ -462,7 +465,7 @@ loki-stack:
               source_labels:
               - __meta_kubernetes_pod_label_openebs_io_logging
               regex: true
-              target_label: mayastor_component
+              target_label: our_component
```
Comment (avishnu) [inline chart/values.yaml]: What is the relevance of this change in all places from mayastor* to our* ?

File: chart/values.yaml (hunk 1 above)
Comment (avishnu) [reply chart/values.yaml:402]: Ok, ignore my above comment. BoltDB seems to be a Grafana term !!!
https://grafana.com/docs/loki/latest/operations/storage/boltdb-shipper/

**Review by Abhinandan-Purkait — APPROVED** [review]

**Review by avishnu — APPROVED** [review]
Thanks !! Lgtm

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors cancel

**Comment by tiagolobocastro** [comment]
bors merge

**Pull request merged**